examples/platforms/efr32/uart.c

```diff
@@ -39,97 +39,89 @@
+static uint16_t              sTransmitLength = 0;
 
-typedef struct RecvBuffer
+typedef struct ReceiveFifo_t
 {
     // The data buffer
-    uint8_t mBuffer[kReceiveBufferSize];
+    uint8_t mBuffer[kReceiveFifoSize];
     // The offset of the first item written to the list.
     uint16_t mHead;
     // The offset of the next item to be written to the list.
     uint16_t mTail;
-} RecvBuffer;
+} ReceiveFifo_t;
 
-static RecvBuffer     sReceive;
-
-otError otPlatUartEnable(void)
-{
-    USART_TypeDef           *usart = USART0;
-    USART_InitAsync_TypeDef init   = USART_INITASYNC_DEFAULT;
+static ReceiveFifo_t sReceiveFifo;
 
-    sReceive.mHead = 0;
-    sReceive.mTail = 0;
-
-    /* Enable peripheral clocks */
-    CMU_ClockEnable(cmuClock_HFPER, true);
-    /* Configure GPIO pins */
-    CMU_ClockEnable(cmuClock_GPIO, true);
-
-    /* Configure GPIO pin for UART TX */
-    /* To avoid false start, configure output as high. */
-    GPIO_PinModeSet(BSP_BCC_TXPORT, BSP_BCC_TXPIN, gpioModePushPull, 1u);
-    /* Configure GPIO pin for UART RX */
-    GPIO_PinModeSet(BSP_BCC_RXPORT, BSP_BCC_RXPIN, gpioModeInput, 1u);
-
-    /* Enable the switch that enables UART communication. */
-    GPIO_PinModeSet(BSP_BCC_ENABLE_PORT, BSP_BCC_ENABLE_PIN, gpioModePushPull, 1u);
-
-    CMU_ClockEnable(cmuClock_USART0, true);
+static void processReceive(void);
 
-    /* Configure USART for basic async operation */
-    init.enable = usartDisable;
-    USART_InitAsync(usart, &init);
+void receiveDone(UARTDRV_Handle_t aHandle, Ecode_t aStatus, uint8_t *aData, UARTDRV_Count_t aCount)
+{
+    // We can only write if incrementing mTail doesn't equal mHead
+    if (sReceiveFifo.mHead != (sReceiveFifo.mTail + 1) % kReceiveFifoSize)
+    {
+        sReceiveFifo.mBuffer[sReceiveFifo.mTail] = aData[0];
+        sReceiveFifo.mTail = (sReceiveFifo.mTail + 1) % kReceiveFifoSize;
+    }
 
-    /* Enable pins at correct UART/USART location. */
-    usart->ROUTEPEN = USART_ROUTEPEN_RXPEN | USART_ROUTEPEN_TXPEN;
-    usart->ROUTELOC0 = (usart->ROUTELOC0 & ~(_USART_ROUTELOC0_TXLOC_MASK | _USART_ROUTELOC0_RXLOC_MASK))
-                       | (_USART_ROUTELOC0_TXLOC_LOC0 << _USART_ROUTELOC0_TXLOC_SHIFT)
-                       | (_USART_ROUTELOC0_RXLOC_LOC0 << _USART_ROUTELOC0_RXLOC_SHIFT);
+    UARTDRV_Receive(aHandle, aData, 1, receiveDone);
+}
 
-    /* Clear previous RX interrupts */
-    USART_IntClear(usart, USART_IF_RXDATAV);
-    NVIC_ClearPendingIRQ(USART0_RX_IRQn);
+otError otPlatUartEnable(void)
+{
+    UARTDRV_Init_t uartInit = USART_INIT;
 
-    /* Enable RX interrupts */
-    USART_IntEnable(usart, USART_IF_RXDATAV);
-    NVIC_EnableIRQ(USART0_RX_IRQn);
+    sReceiveFifo.mHead = 0;
+    sReceiveFifo.mTail = 0;
 
-    /* Finally enable it */
-    USART_Enable(usart, usartEnable);
+    UARTDRV_Init(sUartHandle, &uartInit);
 
-    return OT_ERROR_NONE;
-}
+    UARTDRV_TransmitB(sUartHandle, (uint8_t *) "\nOpenThread\n", 12u);
```

Comment:
Is this line really needed? Other platforms send nothing on otPlatUartEnable.

Reply:
This was left over from my own testing.  Removed!